### PR TITLE
Feature carbonflux

### PR DESCRIPTION
**Pull request by dagibbs22**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Emissions calculated from this tool did not use the forest carbon flux model. 

## What is the new behavior?
- Now users receive emissions, removals, and net flux from the forest carbon flux model.
- Other outputs (not related to emissions) are the same when forest carbon flux output is provided.
- Tool defaults to using forest carbon flux model output because that is the more accurate emissions data.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

None

### CODE REVIEW

File: TreeCoverLossAnalysis.pyt [hunk 1]
```diff
@@ -167,6 +177,7 @@ class TreeCoverLossAnalysis(object):
             tcd_year,
             primary_forests,
             plantations,
+            carbonflux,
```
Comment (thomas-maschler) [inline TreeCoverLossAnalysis.pyt:180]:
```suggestion
            analysis,
```

File: TreeCoverLossAnalysis.pyt [hunk 2]
```diff
@@ -92,6 +92,15 @@ class TreeCoverLossAnalysis(object):
...
 
+        carbonflux = arcpy.Parameter(
+            displayName="Analyze forest carbon flux model (emissions, removals, net flux)",
+            name="carbonflux",
+            datatype="GPBoolean",
+            parameterType="Required",
+            direction="Input",
+        )
+        carbonflux.value = True
+
```
Comment (thomas-maschler) [inline TreeCoverLossAnalysis.pyt:103]:
```suggestion
        analysis = arcpy.Parameter(
            displayName="Analysis type",
            name="analysis",
            datatype="GPString",
            parameterType="Required",
            direction="Input",
        )
        analysis.filter.type = "ValueList"
        analysis.filter.list = ["treecoverloss", "carbonflux_minimal"]
        analysis.value = "treecoverloss"

```

File: TreeCoverLossAnalysis.pyt [hunk 3]
```diff
@@ -203,13 +214,14 @@ class TreeCoverLossAnalysis(object):
         tcd_year = parameters[2].value
         primary_forests = parameters[3].value
         plantations = parameters[4].value
-        master_instance_type = parameters[5].value
-        worker_instance_type = parameters[6].value
-        worker_instance_count = parameters[7].value
-        jar_version = parameters[8].valueAsText
+        carbonflux = parameters[5].value
```
Comment (thomas-maschler) [inline TreeCoverLossAnalysis.pyt:217]:
```suggestion
        analysis = parameters[5].value
```

File: TreeCoverLossAnalysis.pyt [hunk 4]
```diff
@@ -221,13 +233,14 @@ class TreeCoverLossAnalysis(object):
...
             self._load_layer(messages)
-        self._export_wbk(messages)
+        self._export_wkb(messages)
         self._upload_to_s3(messages)
         self._launch_emr("s3://{}/{}/{}".format(self.s3_bucket, self.s3_in_features_prefix, self.tsv_file),
                          tcd,
                          tcd_year,
                          primary_forests,
                          plantations,
+                         carbonflux,
```
Comment (thomas-maschler) [inline TreeCoverLossAnalysis.pyt:243]:
```suggestion
                         analysis,
```

File: TreeCoverLossAnalysis.pyt [hunk 5]
```diff
@@ -327,6 +340,7 @@ class TreeCoverLossAnalysis(object):
             tcd_year,
             primary_forests,
             plantations,
+            carbonflux,
```
Comment (thomas-maschler) [inline TreeCoverLossAnalysis.pyt:343]:
```suggestion
            analysis,
```

File: TreeCoverLossAnalysis.pyt [hunk 6]
```diff
@@ -437,6 +449,11 @@ class TreeCoverLossAnalysis(object):
         if plantations:
             steps[0]["HadoopJarStep"]["Args"].extend(["--contextual_layer", "is__gfw_plantations"])
 
+        if carbonflux:
+            steps[0]["HadoopJarStep"]["Args"].extend(["--analysis", "carbonflux_minimal"])
+        else:
+            steps[0]["HadoopJarStep"]["Args"].extend(["--analysis", "treecoverloss"])
+
```
Comment (thomas-maschler) [inline TreeCoverLossAnalysis.pyt:456]:
```suggestion
        steps[0]["HadoopJarStep"]["Args"].extend(["--analysis", analysis])
```

**Review by thomas-maschler — CHANGES_REQUESTED** [review]
From the readme comment it seems like the carbonflux analysis does the same as the treecover loss analysis, just uses different values for the biomass? Or is a differnt analysis?

Since it is calling a different analysis in Spark, I would prefer to make it a drop down menu. This way, it is easier to add new analysis in the future.

If this only tweaks the tree cover loss analysis and uses different data, why not alter the tree cover loss analysis itself and ditch the old biomass layer?

**Comment by dagibbs22** [comment]
This PR is moot now.

**Pull request closed without merging**